src/agent/src/sev.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2023 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+
+pub fn get_snp_host_data() -> Result<Vec<u8>> {
```

Comment:
Get hostdata field in kata-agent seems arch-specific. I'd suggest to do this in AA side. As it already has the ability to get hardware evidence/quote. TDX is the same.

Reply:
I don't think we should force users to use the CoCo AA. Users should be able to perform remote attestation on their own if they prefer to - based on the arch specific docs.

However, sharing a rust crate between the Kata Agent and AA would sound reasonable to me - not sure if this is what you meant.

If you meant adding some protocol for Policy validation between the Kata Agent and AA, please describe the details when we'll meet. For example, we'd need that kind of protocol to succeed before executing many other Kata Agent API calls - otherwise a not-yet-validated Policy might allow dangerous actions.

Reply:
IMO we should try to make kata-agent arch-independent. Some reasons:

1. CoCoAA has already had the ability of remote attestation related things, so we should try to leverage them rather than re-invention. As you said, users should be able to perform remote attestation on their own if they prefer to. We are refactoring AA to support only have attestation related functionalities like GetAttestationToken and GetEvidence. This means that not only KBS/CoCo-AS can be accessed, but also MAA/Intel TA and other ASes/ways could implement the specific plugins. If we insist to implement the logic inside kata-agent, there will be two copies of the code.
2. If we want to check the hostdata field, it hints that we already enable the confidential computing feature. Furthermore confidential computing feature means that CoCo-AA will be built-in.